class-two-factor-core.php

```diff
@@ -571,6 +571,25 @@ public static function get_provider_for_user( $user = null, $preferred_provider
 		return self::get_primary_provider_for_user( $user );
 	}
 
+	/**
+	 * Get the name of the primary provider selected by the user
+	 * and enabled for the user.
+	 *
+	 * @param WP_User|int $user User ID or instance.
+	 *
+	 * @return string|null
+	 */
+	private static function get_primary_provider_key_selected_for_user( $user ) {
```

Comment:
This is a dedicated helper to get the selected option without attempting to resolve if the method is active or not.